soc/riscv/riscv-privileged/Kconfig

```diff
@@ -28,6 +28,12 @@ config RISCV_HAS_CLIC
 	help
 	  Does the SOC provide support for a Core-Local Interrupt Controller (CLIC).
 
+config RISCV_HAS_CUSTOM_IC
+	bool "Does the SOC provide support for a custom Interrupt Controller (IC)"
+	depends on SOC_FAMILY_RISCV_PRIVILEGED
```

Comment:
why the `depends on`?

Reply:
I was thinking that since soc_common_irq belongs to riscv-privileged path and custom IC path reuses PLIC pathq, then we should restrict this feature to SOC_FAMILY_RISCV_PRIVILEGED.

But looks like you are suggesting entirely independent path for custom IC and we can really have it common for the entire RISC-V, right? Where do you suggest to implement the arch_irq_* wrappers  (over z_soc_*) ?